bin/xbps-checkvers/main.c

```diff
@@ -593,16 +594,27 @@ rcv_check_version(rcv_t *rcv)
 	assert(version.v.s);
 	assert(revision.v.s);
 
-	srcver = strncpy(srcver, pkgname.v.s, pkgname.v.len);
+	rv = xbps_strlcpy(srcver, pkgname.v.s, BUFSIZ);
+	if (rv >= BUFSIZ)
+		return 1;
+
 	if (rcv->installed)
 		rcv->pkgd = xbps_pkgdb_get_pkg(&rcv->xhp, srcver);
 	else
 		rcv->pkgd = xbps_rpool_get_pkg(&rcv->xhp, srcver);
 
-	srcver = strncat(srcver, "-", 1);
-	srcver = strncat(srcver, version.v.s, version.v.len);
-	srcver = strncat(srcver, "_", 1);
-	srcver = strncat(srcver, revision.v.s, revision.v.len);
+	rv = xbps_strlcat(srcver, "-", BUFSIZ);
+	if (rv >= BUFSIZ)
+		return 1;
+	rv = xbps_strlcat(srcver, version.v.s, BUFSIZ);
+	if (rv >= BUFSIZ)
+		return 1;
+	rv = xbps_strlcat(srcver, "_", BUFSIZ);
+	if (rv >= BUFSIZ)
+		return 1;
+	rv = xbps_strlcat(srcver, revision.v.s, BUFSIZ);
+	if (rv >= BUFSIZ)
+		return 1;
```

Comment:
this could probably use on snprintf call but I was overly excited to use strlcat ...